### PR TITLE
feat(api-server): new inspect api

### PR DESCRIPTION
**Pull request by lahabana**
No kumactl for the moment will add in a follow up PR.

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshtcproute/plugin/v1alpha1/plugin_test.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ import (
 	core_xds "github.com/kumahq/kuma/pkg/core/xds"
 	"github.com/kumahq/kuma/pkg/metrics"
 	core_rules "github.com/kumahq/kuma/pkg/plugins/policies/core/rules"
+	_ "github.com/kumahq/kuma/pkg/plugins/policies/meshhttproute"
```
Comment (lukidzi) [inline pkg/plugins/policies/meshtcproute/plugin/v1alpha1/plugin_test.go:19]: Is it needed just for plugin registration?

File: docs/generated/openapi.yaml [hunk 2]
```diff
@@ -19,6 +19,62 @@ paths:
           $ref: '#/components/responses/InternalServerError'
       operationId: get-global-insight
       description: Returns a Global Insight object
+  /meshes/{mesh}/{policyType}/{policyName}/-resources/dataplanes:
```
Comment (lukidzi) [inline docs/generated/openapi.yaml:22]: Why `-resources` is it something specific?

File: pkg/api-server/resource_endpoints.go [hunk 3]
```diff
@@ -438,3 +453,103 @@ func (r *resourceEndpoints) readOnlyMessage() string {
...
+				rest_errors.HandleError(request.Request.Context(), response, err, "failed inspect")
+				return
+			}
+			if err := r.resManager.List(request.Request.Context(), hl, store.ListByMesh(meshName)); err != nil {
+				rest_errors.HandleError(request.Request.Context(), response, err, "failed inspect")
+				return
+			}
+			dependentResources.MeshLocalResources[dependentType] = hl
+		}
+		filter := func(rs model.Resource) bool {
```
Comment (lukidzi) [inline pkg/api-server/resource_endpoints.go:503]: Could we extract this part until the response is prepared to a separate method like `GetAffectedDataplanes`?

File: docs/generated/openapi.yaml (hunk 2 above)
Comment (lahabana) [reply docs/generated/openapi.yaml:22]: Prefix by dash avoids any possible collision with dynamically generated urls from resource paths in the future.
It makes it very clear that this is not an actual entity type

File: pkg/api-server/resource_endpoints.go (hunk 3 above)
Comment (lahabana) [reply pkg/api-server/resource_endpoints.go:503]: I think we could but it'd be a function generator with a bunch of params (dependent resources and policy at least)

**Review by lukidzi — APPROVED** [review]

File: mk/docs.mk [hunk 4]
```diff
@@ -56,7 +56,7 @@ docs/generated/openapi.yaml:
 	rm -rf $(OAPI_TMP_DIR)
 	mkdir -p $(dir $@)
 	mkdir -p $(OAPI_TMP_DIR)/policies
-	for i in $(API_DIRS); do mkdir -p $(OAPI_TMP_DIR)/$$(echo $${i} | cut -d: -f2); cp -r $$(echo $${i} | cut -d: -f1) $(OAPI_TMP_DIR)/$$(echo $${i} | cut -d: -f2); done
+	for i in $(API_DIRS); do mkdir -p $(OAPI_TMP_DIR)/$$(echo $${i} | cut -d: -f2); cp -R $$(echo $${i} | cut -d: -f1) $(OAPI_TMP_DIR)/$$(echo $${i} | cut -d: -f2); done
```
Comment (michaelbeaumont) [inline mk/docs.mk:59]: Is this supposed to be here? Please at least a different commit, ideally PR

File: docs/generated/openapi.yaml (hunk 2 above)
Comment (michaelbeaumont) [reply docs/generated/openapi.yaml:22]: Is this a common open API thing? It seems like we'd be unlikely to dynamically generate a path `../resources` and if we do, isn't that just kind of confusing in spite of the `/-` prefix?

File: pkg/api-server/resource_endpoints_test.go [hunk 5]
```diff
@@ -1037,4 +1038,8 @@ var _ = Describe("Resource Endpoints", func() {
 		Expect(test_metrics.FindMetric(metrics, "api_server_http_requests_inflight")).ToNot(BeNil())
 		Expect(test_metrics.FindMetric(metrics, "api_server_http_response_size_bytes")).ToNot(BeNil())
 	})
+
+	DescribeTable("inspect", func(inputFile string) {
```
Comment (michaelbeaumont) [inline pkg/api-server/resource_endpoints_test.go:1042]: Does the fact that no test files have changed means the output for existing endpoints and resources is exactly the same?

File: mk/docs.mk (hunk 4 above)
Comment (lahabana) [reply mk/docs.mk:59]: Yeah it is I couldn't generate the change openapi without this

File: docs/generated/openapi.yaml (hunk 2 above)
Comment (lahabana) [reply docs/generated/openapi.yaml:22]: It's not anything official in the openAPI spec for sure. But it's common in apis to have a syntax to clearly indicate that things are not a persisted but a computed resource. For example elastic search has a `_search` I think it's quite nice to have APIs that make this difference clear.

File: pkg/api-server/resource_endpoints_test.go (hunk 5 above)
Comment (michaelbeaumont) [reply pkg/api-server/resource_endpoints_test.go:1042]: OK, yeah. you mentioned fixing the inspect API in this PR earlier today so I wasn't sure but I guess that was referring to the new endpoint

File: docs/generated/openapi.yaml (hunk 2 above)
Comment (lahabana) [reply docs/generated/openapi.yaml]: Ok moved to _resources

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**